components/kyma-environment-broker/internal/broker/broker.go

```diff
@@ -14,6 +16,8 @@ const (
 	KymaServiceName = "kymaruntime"
 )
 
+type PlanDefaults func(planID string, platformProvider internal.CloudProvider, parametersProvider *internal.CloudProvider) (*gqlschema.ClusterConfigInput, error)
```

Comment:
importing `inputFactory` to call `GetHyperscalerProviderForPlanID` directly results in import cycle

```
env CGO_ENABLED=0 go build -o kyma-environment-broker ./cmd/broker/main.go
package command-line-arguments
        imports github.com/kyma-project/control-plane/components/kyma-environment-broker/internal/appinfo
        imports github.com/kyma-project/control-plane/components/kyma-environment-broker/internal/broker
        imports github.com/kyma-project/control-plane/components/kyma-environment-broker/internal/process/input
        imports github.com/kyma-project/control-plane/components/kyma-environment-broker/internal/broker: import cycle not allowed
make: *** [/home/wozy/projects/go/src/github.com/kyma-project/control-plane/scripts/generic_make_go.mk:121: build-local] Error 1
```

so I decided to wrap the call in a function and pass that to `ProvisionEndpoint` and `UpdateEndpoint`